anvio/cazymes.py

```diff
@@ -0,0 +1,252 @@
+
+        if not glob.glob(os.path.join(self.cazyme_data_dir, "dbCAN-HMMdb-*.txt")):
+            raise ConfigError(f"It seems you do not have the CAZyme database installed in {self.cazyme_data_dir}, "
+                              f"please run 'anvi-setup-cazymes' download it.")
+
+        # Glob and find what files we have then check if we have them all
+        downloaded_files = glob.glob(os.path.join(self.cazyme_data_dir, '*'))
+        
+        # here we check if the HMM profile is compressed so we can decompress it for next time
+        hmmpress_file_extensions = ["h3f", "h3i", "h3m", "h3p", "txt"]
+        extant_extensions = [os.path.basename(file).split(".")[-1] for file in downloaded_files]
+
+        if hmmpress_file_extensions.sort() != extant_extensions.sort():
+            raise ConfigError("Anvi'o detected that the CAZyme database was not properly compressed with hmmpress. "
+                              "Please 'anvi-setup-cazymes --reset'")
+
+    def process(self):
+        """Search CAZyme HMMs over contigs-db, parse, and filter results"""
+
+        #FIXME: need a smarter way to find the CAZyme HMM file for when users 
+        # want a specific version. This will break if it's not V11
+        hmm_file = os.path.join(self.cazyme_data_dir, "dbCAN-HMMdb-V11.txt")
+
+        # initialize contigs database
+        class Args: pass
+        args = Args()
+        args.contigs_db = self.contigs_db_path
+        contigs_db = dbops.ContigsSuperclass(args)
+        tmp_directory_path = filesnpaths.get_temp_directory_path()
+
+        # get an instance of gene functions table
+        gene_function_calls_table = TableForGeneFunctions(self.contigs_db_path, self.run, self.progress)
+
+        # export AA sequences for genes
+        target_files_dict = {'AA:GENE': os.path.join(tmp_directory_path, 'AA_gene_sequences.fa')}
+        contigs_db.get_sequences_for_gene_callers_ids(output_file_path=target_files_dict['AA:GENE'],
+                                                      simple_headers=True,
+                                                      report_aa_sequences=True)
+
+        # run hmmer
+        hmmer = HMMer(target_files_dict, num_threads_to_use=self.num_threads, program_to_use=self.hmm_program)
+        hmm_hits_file = hmmer.run_hmmer('CAZymes', 'AA', 'GENE', None, None, len(self.function_catalog), hmm_file, None, self.noise_cutoff_terms)
+
+        if not hmm_hits_file:
+            run.info_single("The HMM search returned no hits :/ So there is nothing to add to the contigs database. But "
+                            "now anvi'o will add CAZymes as a functional source with no hits, clean the temporary directories "
+                            "and gracefully quit.", nl_before=1, nl_after=1)
+            shutil.rmtree(tmp_directory_path)
+            hmmer.clean_tmp_dirs()
+            gene_function_calls_table.add_empty_sources_to_functional_sources({'Pfam'})
```

Comment:
This will clean up all the Pfam annotations from a contigs-db if there are no CAZyme hits, and then make it look like CAZymes were never run on that db :)

I understand that you used the template for anvi-run-pfams for this, which makes a lot of sense, but please search for Pfam in this file and make sure there are no residuals.

Reply:
Woops, good catch! I'll sift through the rest of the PR and purge `anvi-run-pfams`!